### PR TITLE
make sure that signed witness is immutable.

### PR DESCRIPTION
**Pull request by christophsturm**
btw I do hate byte arrays

### CODE REVIEW

**Comment by ManfredKarrer** [comment]
We have a wrapper class as well (P2PDataStorage.ByteArray in p2p module) but its only used where needed (e.g. when the bytarray is a key in a hashmap).

**Review by ManfredKarrer — COMMENTED** [review]
utACK

Good find!

File: core/src/test/java/bisq/core/account/sign/SignedWitnessTest.java [hunk 1]
```diff
@@ -10,22 +10,51 @@
...
+    public void testProtoRoundTrip() {
         SignedWitness signedWitness = new SignedWitness(true, witnessHash, witnessHashSignature, arbitrator1Key.getPubKey(), witnessOwner1PubKey, Instant.now().getEpochSecond(), 100);
         assertEquals(signedWitness, SignedWitness.fromProto(signedWitness.toProtoMessage().getSignedWitness()));
+    }
 
+    @Test
+    public void isImmutable() {
+        byte[] signerPubkey = arbitrator1Key.getPubKey();
+        SignedWitness signedWitness = new SignedWitness(true, witnessHash, witnessHashSignature, signerPubkey, witnessOwner1PubKey, Instant.now().getEpochSecond(), 100);
+        byte[] witnessHash = signedWitness.getWitnessHash().clone();
```
Comment (ManfredKarrer) [inline core/src/test/java/bisq/core/account/sign/SignedWitnessTest.java]: Maybe better to use a different name? Now the this.witnessHash and witnessHash are different, which makes the test a bit harder to read IMO.

**Review by ManfredKarrer — APPROVED** [review]
utACK

see Nit comment...

File: core/src/test/java/bisq/core/account/sign/SignedWitnessTest.java (hunk 1 above)
Comment (christophsturm) [reply core/src/test/java/bisq/core/account/sign/SignedWitnessTest.java]: done

**Review by sqrrm — APPROVED** [review]
utACKl

**Pull request merged**